src/bridge/gs_madrona/renderer_gs.py

```diff
@@ -33,50 +29,20 @@ def __init__(
         add_cam_debug_geo=False,
         use_rasterizer=False,
     ):
-        assert rigid is not None, "Rigid body model is required for MadronaBatchRendererAdapter"
+        assert geom_retriever is not None, "GeomRetriever is required for MadronaBatchRendererAdapter"
         assert gpu_id >= 0, "GPU ID must be greater than or equal to 0"
         assert num_worlds > 0, "Number of worlds must be greater than 0"
+        assert num_cameras > 0, "Must have at least one camera for Madrona to work!"
         assert batch_render_view_width > 0, "Batch render view width must be greater than 0"
         assert batch_render_view_height > 0, "Batch render view height must be greater than 0"
 
-        # defaults
-        default_geom_group = 2
-        default_enabled_geom_groups = np.array([default_geom_group])
-
         self.num_worlds = num_worlds
-
-        mesh_vertices = rigid.vverts_info.init_pos.to_numpy()
-        mesh_faces = rigid.vfaces_info.vverts_idx.to_numpy()
-        mesh_vertex_offsets = rigid.vgeoms_info.vvert_start.to_numpy()
-        mesh_face_offsets = rigid.vgeoms_info.vface_start.to_numpy()
-        n_vgeom = rigid.n_vgeoms
-
-        face_start = rigid.vgeoms_info.vface_start.to_numpy()
-        face_end = rigid.vgeoms_info.vface_end.to_numpy()
-        for i in range(n_vgeom):
-            mesh_faces[face_start[i] : face_end[i]] -= mesh_vertex_offsets[i]
-
-        geom_types = np.full((n_vgeom,), 7, dtype=np.int32)
-        geom_groups = np.full((n_vgeom,), default_geom_group, dtype=np.int32)
-        geom_data_ids = np.arange(n_vgeom, dtype=np.int32)
-        geom_sizes = np.ones((n_vgeom, 3), dtype=np.float32)
-        assert num_cameras > 0, "Must have at least one camera for Madrona to work!"
-
-        (
-            geom_mat_ids,
-            mesh_texcoord_num,
-            mesh_texcoord_offsets,
-            mesh_texcoord_data,
-            texture_widths,
-            texture_heights,
-            texture_nchans,
-            texture_data,
-            texture_offsets,
-            material_texture_ids,
-            material_rgba,
-            geom_rgba,
-        ) = self.get_material_data(rigid)
-
+        self.geom_retriever = geom_retriever
+        geom_args_static = self.geom_retriever.retrieve_rigid_meshes_static()
+        for arg_name, arg_value in geom_args_static.items():
+            assert arg_value.flags['C_CONTIGUOUS'], f"{arg_name} data is not continuous."
+            assert arg_value.base is None, f"{arg_name} data is not self-owned."
```

Comment:
Why not just copy on the data locally if necessary instead of raising an exception?

Prefer `arg_value.data.c_contiguous` over direct flag inspection. 

By do you need self-owned property?